pkg/ddc/thin/engine.go

```diff
@@ -61,7 +61,7 @@ func Build(id string, ctx cruntime.ReconcileRequestContext) (base.Engine, error)
 		return nil, fmt.Errorf("engine %s is failed due to type conversion", ctx.Name)
 	}
 
-	isRef, err := CheckReferenceDatasetRuntime(ctx.Client, runtime)
+	isRef, err := CheckReferenceDatasetRuntime(ctx, runtime)
```

Comment:
Maybe in future we can simply check `len(runtime.profileName) == 0` to indicate whether it is a VirtualRuntime or ThinRuntime instead of checking all the dataset mounts.